drivers/spi/spi_sc18is606.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2025, tinyvision.ai
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_sc18is606_spi
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_sc18is606, CONFIG_SPI_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <stdint.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/spi.h>
+#include "spi_context.h"
+
+#define SC18IS606_CONFIG_SPI 0xF0
+#define CLEAR_INTERRUPT      0xF1
+#define IDLE_MODE            0xF2
+#define READ_VERSION         0xFE
+
+#include "spi_sc18is606.h"
+
+struct nxp_sc18is606_data {
+	struct k_mutex bridge_lock;
+	struct spi_context ctx;
+	uint32_t spi_clock_freq;
+	uint8_t spi_mode;
+};
+
+struct nxp_sc18is606_config {
+	const struct i2c_dt_spec i2c_controller;
+};
+
+int nxp_sc18is606_claim(const struct device *dev)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+
+	return k_mutex_lock(&data->bridge_lock, K_FOREVER);
+}
+
+int nxp_sc18is606_release(const struct device *dev)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+
+	return k_mutex_unlock(&data->bridge_lock);
+}
+
+int nxp_sc18is606_transfer(const struct device *dev, const uint8_t *tx_data, uint8_t tx_len,
+			   uint8_t *rx_data, uint8_t rx_len)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+	const struct nxp_sc18is606_config *info = dev->config;
+	int ret;
+
+	ret = k_mutex_lock(&data->bridge_lock, K_FOREVER);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (tx_data != NULL) {
+		ret = i2c_write(info->i2c_controller.bus, tx_data, tx_len,
+				info->i2c_controller.addr);
+		if (ret) {
+			LOG_ERR("SPI write failed: %d", ret);
+			return ret;
+		}
+	}
+
+	if (rx_data != NULL) {
+		/* What is the time */
+		k_timepoint_t end;
+
+		/*Set a deadline in a second*/
```

Comment:
If hesitating between the two comments style, Zephyr mostly uses `/*<space><text><space>*/`.

P.S.: But it seems like for different languages (looking at code with chinese characters on gitee.com for instance), it's `/*<text>*/` like you did. :)